### PR TITLE
Update PRUserial API

### PR DESCRIPTION
**Pull request by xresende**

### CODE REVIEW

File: client/PRUserial485/PRUserial485.py [hunk 1]
```diff
@@ -161,101 +135,157 @@ def socket_communicate(conn_port, data_queue):
...
     else:
         return None
 
 
-def PRUserial485_write(data=[], timeout=0):
+def PRUserial485_write(data=None, timeout=0):
     """Envia dados através da interface serial."""
     # Payload: TIMEOUT (4 bytes) + DATA (len(DATA) bytes)
+    if data is None:
+        data = []
```
Comment (pnallin) [inline client/PRUserial485/PRUserial485.py]: Lines 239/240 not needed if defining function with "def PRUserial485_write(data=[], timeout=0)"

File: client/PRUserial485/PRUserial485.py [hunk 2]
```diff
@@ -161,101 +135,157 @@ def socket_communicate(conn_port, data_queue):
...
+    payload = _c.COMMAND_PRUserial485_address
+    command, payload_recv = _send_communication_data(payload)
+    if command == ord(_c.COMMAND_PRUserial485_address) and \
+       len(payload_recv) == 1:
+        return ord(payload_recv)
+    else:
+        return None
+
+
+# --- Deprecated functions ---
```
Comment (pnallin) [inline client/PRUserial485/PRUserial485.py]: These functions **are not deprecated** at all! They are implemented on server (BBB) side and working as expected, so they can be added to the library.

The difference here is that IOCs are not using them anymore.

File: client/PRUserial485/PRUserial485.py [hunk 3]
```diff
@@ -14,62 +14,34 @@
...
+# BeagleBone IP
+BBB_IP = None
 
 
+# Queues
+QUEUE_GENERAL = Queue()
+QUEUE_RW = Queue()
 
+# Dictionary for COMM_RESPONSE
+COMM_RESPONSE = {}
```
Comment (pnallin) [inline client/PRUserial485/PRUserial485.py]: Any specific reason for renaming these variables?

File: client/PRUserial485/PRUserial485.py [hunk 4]
```diff
@@ -161,101 +135,157 @@ def socket_communicate(conn_port, data_queue):
...
     return command, payload_recv
 
 
+# --- PRUserial485 API ---
+
+
+def PRUserial485_set_bbb_ip_address(ip_address):
+    """Define beaglebone IP address."""
+    global BBB_IP
+    BBB_IP = ip_address
```
Comment (pnallin) [inline client/PRUserial485/PRUserial485.py]: if "10.128." in ip_address and \
----len(ip_address.split(".")) == 4:
----BBB_IP = ip_address


Garantir que seja um IP válido e dentro da rede de controle

File: client/PRUserial485/PRUserial485.py (hunk 2 above)
Comment (xresende) [reply client/PRUserial485/PRUserial485.py]: @pnallin ,

I did not know they were used by another client other than power supply IOCs. That's why I thought they could be discarded. if not, the library could be simplified by discarding them and thus more easily optimized. This was my rational. 

Are they going to be used by another client? If they are, I will just delete de comment line.

File: client/PRUserial485/PRUserial485.py (hunk 3 above)
Comment (xresende) [reply client/PRUserial485/PRUserial485.py]: We use visual studio code editor which has an audit software called pylama. Pylama suggests using capital letters for module variables, according to PEP008, https://www.python.org/dev/peps/pep-0008/

But this is just a recommendation that avoids visual studio marking issues along the code. should we keep the original names?

File: client/PRUserial485/PRUserial485.py (hunk 4 above)
Comment (xresende) [reply client/PRUserial485/PRUserial485.py]: done!

File: client/PRUserial485/PRUserial485.py (hunk 1 above)
Comment (xresende) [reply client/PRUserial485/PRUserial485.py]: It is considered a good practice not using mutables as default function arguments. Visual Studio code (code editor we use) keeps warning about it. That's why I changed it.

https://stackoverflow.com/questions/366422/what-is-the-pythonic-way-to-avoid-default-parameters-that-are-empty-lists

https://docs.python.org/3/reference/compound_stmts.html#function-definitions

File: client/PRUserial485/PRUserial485.py (hunk 2 above)
Comment (xresende) [reply client/PRUserial485/PRUserial485.py]: @pcnallin,

In the lattest commit, that transformed the client API from module functions to class methods, I included these functions.

**Pull request merged**